packages/headless/src/modules/auth/repositories/index.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@nestjs/common';
+import { User } from 'src/entity/user';
+import { IAuthDatabase } from './auth.database.interface';
+@Injectable()
+export class AuthRepository {
+  constructor(private readonly db: IAuthDatabase) {}
+
+  async createUser(user: User): Promise<User> {
+    return await this.db.createUser(user);
+  }
+
+  async findUser(username: string): Promise<User> {
+    return await this.db.findUser(username);
+  }
```

Comment:
I think create user, find user should be part of user repository, and auth service may depend on user repository for this